cpg_utils/flows/status.py

```diff
@@ -0,0 +1,192 @@
+"""
+Metamist wrapper to report analysis progress.
+"""
+
+from textwrap import dedent
+from abc import ABC, abstractmethod
+
+from cpg_utils import Path
+from cpg_utils.hail_batch import image_path, command
+from hailtop.batch.job import Job
+from hailtop.batch import Batch, Resource
+
+from .targets import Target, Sample
+from .metamist import get_metamist, AnalysisStatus, MetamistError
+
+
+class StatusReporter(ABC):
+    """
+    Status reporter
+    """
+
+    @abstractmethod
+    def add_updaters_jobs(
+        self,
+        b: Batch,
+        output: str | Path | Resource | dict[str, Path | Resource],
+        analysis_type: str,
+        target: Target,
+        jobs: list[Job] | None = None,
+        prev_jobs: list[Job] | None = None,
+        meta: dict | None = None,
+    ) -> list[Job]:
+        """
+        Add Hail Batch jobs that update the analysis status.
+        """
+
+    @abstractmethod
+    def create_analysis(
```

Comment:
@illusional I wonder if you had any thoughts on this, based on our chat yesterday.

Reply:
Not fully formed yet, I'll try to summarise all my thoughts into a doc soon based on chatting to everyone.